### PR TITLE
Publish Java API client to Sonatype repository

### PR DESCRIPTION
**Pull request by arielshaqed**
The OpenAPI generator writes its own `pom.xml` file, and `pom.xml` files are non-composable.
Instead request that it use a *parent* `pom.xml`, configure signing and deployment there (in
an ungenerated file which we can control entirely).

To deploy, run `mvn deploy` inside `clients/java` (*not* `clients`!).

### CODE REVIEW

**Comment by arielshaqed** [comment]
(Already published 0.1.0 with this, sorry-not-sorry)

File: clients/hadoopfs/pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
         <dependency>
             <groupId>io.lakefs</groupId>
             <artifactId>api-client</artifactId>
-            <version>0.1.0</version>
+            <version>0.40.3-RC.0</version>
```
Comment (talSofer) [inline clients/hadoopfs/pom.xml:33]: This should be 0.1.0 right?

Comment (arielshaqed) [reply clients/hadoopfs/pom.xml]: Nope, but I bumped it to be the release version.  (We may revisit this decision later).

**Review by arielshaqed — COMMENTED** [review]
Thanks!  PTAL...

File: clients/hadoopfs/pom.xml [hunk 2]
```diff
@@ -62,7 +62,7 @@
         <dependency>
             <groupId>io.lakefs</groupId>
             <artifactId>api-client</artifactId>
-            <version>0.1.0</version>
+            <version>0.40.3</version>
```
Comment (talSofer) [inline clients/hadoopfs/pom.xml:65]: version is incorrect

File: clients/pom.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    </developer>
+  </developers>
+
+  <scm>
+    <url>https://github.com/treeverse/lakefs</url>
+    <connection>scm:git:git://github.com/treeverse/lakefs.git</connection>
+    <developerConnection>scm:git:ssh://git@github.com/treeverse/lakefs.git</developerConnection>
+  </scm>
+
+  <profiles>			<!-- to deploy (signed), must run "gpg -Prelease deploy"-->
```
Comment (talSofer) [inline clients/pom.xml:36]: I would expect the deploy command to use the profile id you defined below. i.e. "gpg -Ptreeverse-signing deploy"

**Review by talSofer — APPROVED** [review]

File: clients/pom.xml (hunk 3 above)
Comment (arielshaqed) [reply clients/pom.xml]: Indeed you can do that. Not sure where this profile selection is missing, so will add it when I discover.

Thanks!

**Review by arielshaqed — COMMENTED** [review]
Thanks!
Pulling. . .

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Pull request merged**